include/swift/AST/DiagnosticsSIL.def

```diff
@@ -476,6 +476,11 @@ NOTE(autodiff_cannot_differentiate_through_inout_arguments,none,
      "cannot differentiate through 'inout' arguments", ())
 NOTE(autodiff_class_member_not_supported,none,
      "differentiating class members is not yet supported", ())
+// TODO(TF-642): Remove when `partial_apply` works with `@differentiable`
+// functions.
+NOTE(autodiff_cannot_index_subset_thunk_partial_applied_orig_fn,none,
```

Comment:
What's the verb here? Cannot index?

Reply:
"Cannot thunk." Let me know if you have a better name idea.

Reply:
Renamed in https://github.com/apple/swift/pull/26292/commits/6e170f2228ce1f0e13b3e1d5dc988d4f8a82580f to `autodiff_cannot_param_subset_thunk_partially_applied_orig_fn`, which is more clear.